src/wavesurfer.js

```diff
@@ -401,6 +401,9 @@ var WaveSurfer = {
             ajax.on('success', callback),
             ajax.on('error', function (e) {
                 my.fireEvent('error', 'XHR error: ' + e.target.statusText);
+            }),
+            ajax.on('destroy', function (e) {
```

Comment:
The `ajax` object doesn't have a `destroy` event. It should listen to the wavesurfer's `destroy` event. It should also unsubscribe from it on success or error (of Ajax).

Also use `once` instead of `on`.